src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -83,6 +88,39 @@ def sql(self) -> str:
     def __str__(self) -> str:
         return self.pretty_name
 
+    @property
+    def plan_node_category(self) -> PlanNodeCategory:
+        return PlanNodeCategory.OTHERS
+
+    @property
+    def individual_node_complexity(self) -> Dict[str, int]:
+        """Returns the individual contribution of the expression node towards the overall
```

Comment:
so what have been little bit confusing is how you overwrite the individual_node_complexity and  cumulative_node_complexity, for example, in join, i see you overwritten the individual_node_complexity where you were also counting the complexity of the join_cond etc. However, when i look at star expression it seems we overwritten the cumulative_node_complexity by accumulating all expressions.

So this leaves the definition of those two function kind of vague, how about this, let's say individual_node_complexity is the complexity of all nodes except the children. the cumulative_node_complexity for Expression class is the sum of individual complexity and all children's complexity, idealy all class that extends the Expression just need to overwrite the individual complexity correctly. Also let's leave a comment here to make it clear.
Or you can define it in the opposite way, where individual complexity simply represent the complexity of the current node, like what kind of node category it belongs to. in other words, the join should also overwrite the accumulative complexity. 
Either way on your preference, but let's make it consist, and make the function definition clear with comment

Reply:
done